File: quiz/week10/equi_filter.py
#!/usr/bin/python3
import sys

DICT = {}
result = ""
token = False
for line in sys.stdin.readlines():
    line=line.strip('\n')
    words = line.split(" ")
    for word in words:
        for letter in word.lower():
            if(letter in DICT.keys()):
                DICT[letter] = DICT[letter] + 1
            else:
                DICT[letter] = 1
        if len(set(DICT.values())) == 1:
            token = True
            result = result + word + " "
        DICT = {}
    if token:
        result = result.strip()
        result = result + '\n'
    else:
        result = result + '\n'
    token = False
print(result,end='')

File: quiz/week10/equi_filter.pl
#!/usr/bin/perl -w
$result = '';
sub isEqui{
	my %hash = ();
	my $value = 0;
	my $word = $_[0];
	@letters = split(//,$word);
	foreach $letter(@letters){
		$hash{$letter}++;
	}	
	
	@keys = keys %hash;
	foreach $key(@keys){
		if($value == 0){
			$value = $hash{$key};
			next;
		}
		if($value != $hash{$key}){
			return 0;
		}
	}	
	return 1;
	
}
while($line = <STDIN>){
	chomp $line;
	@words = split(/\s+/,$line);
	$match = 0;
	foreach $word(@words){
		$tmp = $word;
		if(isEqui(lc($tmp))){
			$result = $result.$word." ";
			$match = 1;	
		}		
	}
	if($match){
	$result =~ s/\s$//;
	$result = $result."\n";
	}else{
	$result = $result."\n";
	}
}

print $result;

